.github/integrate.yml

```diff
@@ -0,0 +1,30 @@
+name: ⚡ CI/CD 🚀
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  integration-deployment:
+    name: ⚡ Integration & Deployment 🚀
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v2
+      - name: Setup 🏗️
+        run: npm install
+      - name: Linting Tests 🧶
+        run: npm run lint
+      - name: Unit Tests 🎉
+        run: npm run test
+      - name: Build React Project 🛠️
+        run: npm run build
+      - name: Build Storybook 📖
+        run: npm run build-storybook
+      - name: Deploy to GitHub Pages 📄
+        uses: JamesIves/github-pages-deploy-action@v4
+        with:
+          branch: gh-pages
+          folder: build
```

Comment:
Integration should run whenever someone pushes to their branch that has a pull request and deployment should only run when a branch is merged into master. Integration and deployment need to be separate, otherwise the development server will be hosting branches that are in review.